### PR TITLE
Icon component

### PR DESCRIPTION
**Pull request by codingMustache**
Refactor: Made component for parsing Icons with links attached

 ## Feat of Component 

1. pass icon name + link to be attached return icon with link
2. can be extended with other icons in the return statement
3. Can be reused in cases of needing an Icon from Ant Design

### CODE REVIEW

**Comment by colinwilliams91** [comment]
there might be a small
merge conflict between this PR and [mine](https://github.com/Nola-Devs/Nola-Dev-Site/pull/99)

i added a case for twitter with the respective ant-icon using your old implementations syntax, fwiw

**Comment by codingMustache** [comment]
I don't think it will cause a merge conflict as long as you merge your's 1st. I saw that you added a twitter icon and the new component reflects that in the parser.

File: app/[slug]/page.tsx [hunk 1]
```diff
@@ -74,78 +70,11 @@ export default function Group() {
                 <p>{e.name}</p>
                 <div className="flex items-center self-center gap-2">
                   {e.links?.map((e: urlDTO, i: number) => {
```
Comment (colinwilliams91) [inline app/[slug]/page.tsx:72]: not breaking, but difficult readability: iterating over "e" when items inside are also "e"?

File: app/[slug]/page.tsx [hunk 2]
```diff
@@ -74,78 +70,11 @@ export default function Group() {
...
-                            <LinkOutlined />
-                          </a>
-                        );
-                        break;
-                    }
-                    return icon;
-                  })}
+                    const icon = Object.keys(e)[0];
+                    const linkToIcon = Object.values(e)[0];
+                    return <IconParser icon={ icon } linkToIcon={ linkToIcon } key={ linkToIcon + i } /> ;
```
Comment (colinwilliams91) [inline app/[slug]/page.tsx:75]: 👍 this is cool

**Review by colinwilliams91 — APPROVED** [review]

**Pull request merged**